name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/KrakenQuoteFeed.java

```diff
@@ -0,0 +1,178 @@
+package name.abuchen.portfolio.online.impl;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+
+import name.abuchen.portfolio.Messages;
+import name.abuchen.portfolio.model.Exchange;
+import name.abuchen.portfolio.model.LatestSecurityPrice;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.SecurityPrice;
+import name.abuchen.portfolio.online.QuoteFeed;
+
+public final class KrakenQuoteFeed implements QuoteFeed
+{
+
+    public static final String ID = "KRAKEN"; //$NON-NLS-1$
+
+    private static final String QUOTE_URL = "https://api.kraken.com/0/public/OHLC?pair={0}&since={1}&interval=1440";
+
+    @Override
+    public String getId()
+    {
+        return ID;
+    }
+
+    @Override
+    public String getName()
+    {
+        return Messages.LabelKraken;
+    }
+
+    @Override
+    public boolean updateLatestQuotes(Security security, List<Exception> errors)
+    {
+        if (security.getTickerSymbol() == null)
+        {
+            errors.add(new IOException(MessageFormat.format(Messages.MsgMissingTickerSymbol, security.getName())));
+            return false;
+        }
+
+        List<LatestSecurityPrice> prices = getHistoricalQuotes(security, LocalDate.now(), errors);
```

Comment:
The latest quotes are also fetched via the same API as the historical quotes. Which can lead to two calls to the API (one for latest and for historical) which essentially retrieve the same. Is there a rate limit on the API? If so, it might make sense to cache the results for 5 minutes and avoid the second call.